toc.js

```diff
@@ -1,106 +1,82 @@
-
-// Use document.querySelector to find the <aside> element with the id
-// "table-of-contents". Store this in a variable called `toc`.
-
-// Sanity check: console.log the `toc` variable
-
-// Append the `ol` containing the table of contents list to `toc`.
-
-// Use document.createElement to generate an h2 element with the text 
-// "Table of Contents"
-
-// Use insertBefore to add the heading to `toc` *before* the `ol` containing the
-// list of internal links.
-
-// Sanity check: console.log `toc`
-
-
-
-//
-// 2) Make it collapsible
-//
-
-// use addEventListener to listen for 'click' events on the "Table of Contents"
-// h2 element.
-
-// When the Table of Contents heading is clicked, use .classList.toggle() to add
-// or remove the 'collapsed' class name on the `ol` list of links.
-
-// Add a style.css file with a rule to hide (using `display: none;`) elements
-// that have the `collapsed` class name.
-
-
-
-
-
-
-//
-// 3) Make the TOC a nested list: (h3s are nested inside h2s) 
-//
-
-// Update the code generating the table of contents list above to make it a 
-// nested list like so:
-//
-// <ol>
-//   <li>
-//     <a href="#description" rel="internal">Description</a>
-//     <ol>
-//       <li>
-//         <a href="#create-zoo-js-with-the-animals-array" rel="internal">Create zoo.js with the animals array</a>
-//       </li>
-//       <li>
-//         <a href="#sort-and-display-the-animal-gallery" rel="internal">Sort and display the animal gallery</a>
-//       </li>
-//       <li>
-//         <a href="#display-the-featured-animal" rel="internal">Display the featured animal</a>
-//       </li>
-//       <li>
-//         <a href="#map-sort-and-display-animal-ages" rel="internal">Map, sort, and display animal ages</a>
-//       </li>
-//     </ol>
-//   <li>
-//     <a href="#animals" rel="internal">Animals</a>
-//   </li>
-//   ...
-// </ol>
+// toc.js
+// Auto-generates a Table of Contents
+
+function autoToC() {
+    
+  // Use document.querySelector all to find all the headings in the article 
+  // **except** for the h1 (i.e. h2, h3, h4, h5, h6)
+  var headings = document.querySelectorAll("h2, h3, h4, h5, h6");
+
+
+  // Use document.createElement to create a new `ol` element that will hold the
+  // table of contents list
+  var tocList = document.createElement('ol');
+
+
+  // Loop through the headings using a 'for' loop, and create an ordered list of 
+  // links to the different sections of the document.
+  for (var i = 0; i < headings.length; i++) {
+      var heading = headings[i];
+      var ul = document.createElement('ul');
+
+      // Finds h3 element that has an h2 element before it
+      if (heading.tagName === 'H3' && headings[i - 1].tagName === 'H2') {
+          var h = i;
+        
+          // adds h3 elements to ul until an h2 element is reached
+          while (headings[h].tagName === 'H3') {
+              var a = document.createElement('a');
```

Comment:
Lines 28 through 33 are identical to lines 41 through 47. It would be good to define and use a function to create those elements based on the heading element passed in.

Something like:

```js
createListItem(heading) {
    var a = document.createElement('a');
    a.href = "#" + heading.id;
    // ...
    var li = document.createElement('li');
    // ...
    return li;
}
```